### PR TITLE
UI and model improvements

### PR DESCRIPTION
**Pull request by aawad1**
Added buttons to switch from coordinates and place name, crated new view for coordinates.

Added navbar, history, about, and did the parsing of query

### CODE REVIEW

File: env copy/pyvenv.cfg
Comment (mehmedkadric) [inline env copy/pyvenv.cfg:1]: This file doesn't belong to github

File: env copy/Scripts/Activate.ps1
Comment (mehmedkadric) [inline env copy/Scripts/Activate.ps1:1]: These files don't belong to github (env copy, env folders, scripts folder and all files in it)

File: main/migrations/0002_geolocatorstats_latitude_geolocatorstats_longitude.py
Comment (mehmedkadric) [inline main/migrations/0002_geolocatorstats_latitude_geolocatorstats_longitude.py:1]: This file doesn't belong to github (migrations folder in general)

File: main/forms.py [hunk 1]
```diff
@@ -4,8 +4,15 @@
 class GeolocatorStatsForm(forms.ModelForm): #indicira da se radi o formi kreiranoj na osnovu modela
     class Meta:
         model=GeolocatorStats
-        fields=["query"]
+        fields=["query", "latitude", "longitude"]
```
Comment (mehmedkadric) [inline main/forms.py:7]: Why lat lon?

File: main/static/css/my_styles.css [hunk 2]
```diff
@@ -0,0 +1,87 @@
+/* your_styles.css */
```
Comment (mehmedkadric) [inline main/static/css/my_styles.css:1]: Common name for main css file is `style.css`.

File: main/static/images/pexels-aksonsat-uanthoeng-10788502.jpg
Comment (mehmedkadric) [inline main/static/images/pexels-aksonsat-uanthoeng-10788502.jpg:1]: Read about handling static files with Django. Did you use `python manage.py collectstatic` command? (E.g. set DEBUG=False and see how your app looks like). Ping me in private if you need more info on this.

File: main/templates/coordinates.html [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        <a href="/" style="color: inherit; text-decoration: none;">GeoLocator</a>
+    </h1>
+</div>
+
+<div class="col-12 d-flex justify-content-start">
+    <button type="button" class="btn btn-sm btn-primary" onclick="redirectToName()">Name</button>
+</div>
+
+<script>
+    function redirectToName() {
```
Comment (mehmedkadric) [inline main/templates/coordinates.html:16]: Perhaps you can use `href` tag instead having this function? Something like `href={% url 'homepage' %}` (not sure about syntax).

File: main/templates/history.html [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+            <th scope="col text-light">ID</th>
+            <th scope="col text-light">Query</th>
+            <th scope="col text-light">Latitude</th>
+            <th scope="col text-light">Longitude</th>
+        </tr>
+    </thead>
+    <tbody>
+        {% for item in history %}
+        <tr>
+            <td class="text-light">{{item.id}}</td>
```
Comment (mehmedkadric) [inline main/templates/history.html:18]: Are you sure we don't want to use for loop counter here? Just an idea: use pagination.

File: main/views.py [hunk 5]
```diff
@@ -1,23 +1,77 @@
...
+                context = {"form": form}  
+                return render(request,'locator.html', context=context)
+            if(len(result) == 1):
+                print(resp)
+                fwg = forward_geocode(resp)
+                if(not fwg):
+                    context = {"form": form}  
+                    return render(request,'locator.html', context=context)
+                else:
+                    form.instance.query = fwg[0].get('name')
```
Comment (mehmedkadric) [inline main/views.py:32]: We talked about this. Create 'manually' a new record in the database with all necessary attributes.

File: main/views.py [hunk 6]
```diff
@@ -1,23 +1,77 @@
...
 from .forms import GeolocatorStatsForm
+from .forms import GeolocatorStatsFormHistory
 from .models import GeolocatorStats
 import requests
+import re
 
+# HOW TO DEAL WITH ERROR MESSAGES 
+history = []
+#navbar, history, about, home
 def index(request):
```
Comment (mehmedkadric) [inline main/views.py:12]: Consider having multiple functions for handling different cases.
if condition1:
  function1()
 elif condition2:
  function2()
else:
  functionElse()

File: main/views.py [hunk 7]
```diff
@@ -1,23 +1,77 @@
...
+            latitude, longitude = map(float, parts)
+            return [latitude,longitude]
+            # Check if latitude and longitude are within the valid range
+            if -90 <= latitude <= 90 and -180 <= longitude <= 180:
+                return [latitude, longitude]
+            else:
+                return [200,200]
+        else:
+            return [input_str]
+    except ValueError:
```
Comment (mehmedkadric) [inline main/views.py:73]: Investigate how to handle exceptions in Python. Check for function decorators and error handling (Initial resources: ChatGPT, https://www.geeksforgeeks.org/error-handling-in-python-using-decorators/ ).

File: main/views.py [hunk 8]
```diff
@@ -26,6 +80,46 @@ def forward_geocode(address, format='json'):
     }
 
     response = requests.get(base_url, params=params)
+    
     data = response.json()
+    print(data)
+    return data
+######################################
```
Comment (mehmedkadric) [inline main/views.py:87]: What is the purpose of this line?

File: main/views.py [hunk 9]
```diff
@@ -26,6 +80,46 @@ def forward_geocode(address, format='json'):
...
+def coordinates(request):
+    # Your view logic goes here
```
Comment (mehmedkadric) [inline main/views.py:89]: What is the purpose of this line?

File: main/views.py [hunk 10]
```diff
@@ -26,6 +80,46 @@ def forward_geocode(address, format='json'):
...
+    
+    if request.method == "POST":
+        form = GeolocatorStatsFormLL(request.POST)
+        if form.is_valid():
+            latitude = form.cleaned_data['latitude']
+            longitude = form.cleaned_data['longitude']
+            form.save()
+            return render(request, 'coordinates.html', {"form": form, "place_name": reverse_geocode(latitude, longitude)[0]})
```
Comment (mehmedkadric) [inline main/views.py:97]: Maybe define context variable first (too much code in a single line here).

**Review by mehmedkadric — CHANGES_REQUESTED** [review]

File: main/views.py (hunk 8 above)
Comment (aawad1) [reply main/views.py:87]: just for me to see what is "data"

**Review by mehmedkadric — APPROVED** [review]

**Pull request closed without merging**